e2e_playwright/st_plotly_chart_test.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import pytest
+from playwright.sync_api import Page, expect
+
+from e2e_playwright.conftest import ImageCompareFunction
+
+
+# Only do chromium as this can create a lot of screenshots
+# there should be no differences between chrome and safari and firefox
```

Comment:
I think we should add one more test function that tests only a selection of maybe around 3 charts on all browsers so that we have at least some coverage for plotly on all browser engines.

Reply:
E.g. you could split it up into one function testing basic charts on all browsers (line, bar, ...) and one function testing some more advanced charts only on chromium.

Reply:
Sounds good. Will add 2.